src/routes/index.ts

```diff
@@ -94,11 +94,33 @@ const handleCsvFile = async (
   }
 };
 
-const handleJsonFile = (file: Express.Multer.File): UploadResponse => {
+const handleJsonFile = async (
+  file: Express.Multer.File,
+  bucket: string,
+  region: string
+): Promise<UploadResponse> => {
   if (!validateJsonFile(file.buffer)) {
     return createErrorResponse('INVALID_JSON_FORMAT', 'Invalid JSON file format');
   }
-  return createSuccessResponse('JSON_VALID', 'JSON file is valid - Future implementation');
+
+  const jsonString = file.buffer.toString().replace(/\n/g, '').replace(/\r/g, '');
+  const fileUrl = await saveToTmp(Buffer.from(jsonString), file.originalname);
+  try {
+    const uploadResult = await uploadToMinio(
+      fileUrl,
+      file.originalname,
+      bucket,
+      file.mimetype
+    );
+    await fs.unlink(fileUrl);
+
+    return uploadResult.success
+      ? createSuccessResponse('UPLOAD_SUCCESS', uploadResult.message)
+      : createErrorResponse('UPLOAD_FAILED', uploadResult.message);
+  } catch (error) {
+    logger.error('Error uploading file to Minio:', error);
+    throw error;
+  }
```

Comment:
Can we have common logic to remove the deduplication for the handle file functionality?